tests/test_version_providers.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from textwrap import dedent
+from typing import TYPE_CHECKING, Iterator, Optional
+
+import pytest
+
+from commitizen.config.base_config import BaseConfig
+from commitizen.exceptions import VersionProviderUnknown
+from commitizen.providers import (
+    CargoProvider,
+    CommitizenProvider,
+    ComposerProvider,
+    NpmProvider,
+    Pep621Provider,
+    PoetryProvider,
+    ScmProvider,
+    get_provider,
+)
+from tests.utils import create_file_and_commit, create_tag
+
+if TYPE_CHECKING:
+    from pytest_mock import MockerFixture
+
+
+@pytest.fixture
+def chdir(tmp_path: Path) -> Iterator[Path]:
+    cwd = Path()
+    os.chdir(tmp_path)
+    yield tmp_path
+    os.chdir(cwd)
+
+
+def test_default_version_provider_is_commitizen_config(config: BaseConfig):
+    provider = get_provider(config)
+
+    assert isinstance(provider, CommitizenProvider)
+
+
+def test_raise_for_unknown_provider(config: BaseConfig):
+    config.settings["version_provider"] = "unknown"
+    with pytest.raises(VersionProviderUnknown):
+        get_provider(config)
+
+
+def test_commitizen_provider(config: BaseConfig, mocker: MockerFixture):
```

Comment:
nitpick: I notice we have some duplicate code for providers other than scm-provider. Should we use `parametrize` to de duplicate? I'm good with the current implementation. But it would be even better if we could do that :)

Reply:
Done, I parametrized the file providers tests